EE10 and MP6 GA

tevans78 opened the pull request:
Move EE10 and MP6 to GA

Fixes #24165

tevans78 committed:
EE10 GA

tevans78 commented:
#build

tevans78 committed:
GA MicroProfile 6.0

tevans78 committed:
mpTelemetry-1.0 supersedes all open tracing features

tevans78 commented:
#build

tevans78 commented:
Combined PB: https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_3AACsGdWEe2FXtyvmy2tqA

PB was green and also passed initial Liberty repo tests

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/public/restfulWSLogging-3.0/io.openliberty.restfulWSLogging-3.0.feature:
```diff
@@ -9,6 +9,6 @@ Subsystem-Name: Jakarta RESTful Web Services 3.0 with HTTP logging
 -bundles=\
  io.openliberty.restfulWS.internal.logging.filter
-kind=beta
+kind=ga
```
This should not go to GA.

tevans78 replied:
removed

jhanders34 commented:
I also wonder about the supersedes on whether that is completely right to be all of the mpOpenTracing and opentracing features to have supersede in them.  You cannot use mpTelemetry with EE7, 8 or 9 or MP 1 through 5.  It is a documentation only thing I believe or possibly puts out a message in the messages.log.

tevans78 commented:
#build

tevans78 commented:
#build

tevans78 commented:
Combined PB: https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_fmLpoLaUEe2r6re_Zvbagg

tevans78 commented:
Another combined PB: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_nB8o0LdeEe2r6re_Zvbagg

tevans78 commented:
combined PB: https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_bdhSkLg-Ee2r6re_Zvbagg

tevans78 commented:
#build

tevans78 commented:
#build

tevans78 commented:
Combined PB: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_RRUKALhOEe2r6re_Zvbagg

jhanders34 commented on dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/config/liberty/CDI12ContainerConfig.java:
```diff
@@ -48,8 +47,7 @@ protected void activate(ComponentContext compcontext, Map<String, Object> proper
         }
-        //TOOD the ProductInfo.getBetaEdition() check should be removed before GA
-        if (TraceComponent.isAnyTracingEnabled() && tc.isInfoEnabled() && ProductInfo.getBetaEdition()) {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isInfoEnabled()) {
```
@tevans78 is on vacation.  He put this check in.  Not sure if this is what was intended or not.  Do we want this message to come out every time or only when the customer enables some sort of fine, finer or finest trace?  If the former the isAnyTracingEnabled should be removed.  @Emily-Jiang or @benjamin-confino do you know about what the conversations where related to this change?

Emily-Jiang replied:
I am not sure. @Azquelt do you know anything about this?

Azquelt replied:
Good spot @jhanders34, this looks like a bug. The message should be output whether tracing is enabled or not. The intention is thankfully listed [in the UFO](https://ibm.ent.box.com/file/925747728313) (slide 45).

It looks like the test which checks this message doesn't intentionally set any trace specification, but when it runs it has `*=info:com.ibm.ws.runtime.update.*=all:logservice=detail` which I guess is added automatically.

jhanders34 replied:
Thanks Andrew.  I have updated the code to remove the if statement completely.

jhanders34 commented:
#build

jhanders34 committed:
EE10 Update product code, meta type, and tests to remove beta settings

- Remove beta checking in CDI code for putting out a message.
- Update Jakarta EE 10 changes to metatype to no longer have
ibm:beta="true" or internal setting for new attributes.
- Update tests to no longer set the beta flag in their server configs
and update test logic appropriately.

jhanders34 committed:
MP6 Update product code, meta type, and tests to remove beta settings

- Update MP JWT beta checking code to no longer check beta before doing
certain function.
- Update MP 6 changes to metatype to no longer have ibm:beta="true"
setting for new attributes.
- Update tests to no longer set the beta flag in their server configs
and update test logic appropriately.
- Update metatype description to not be longer than 50 characters.

jhanders34 commented:
#libby #build

helyarp reviewed (commented):
Looks good

posmith commented:
L2 message review not required

pmd1nh commented:
reviewed dev/com.ibm.ws.webcontainer/resources/OSGI-INF/metatype/metatype.xml

pmd1nh reviewed (dismissed):
dev/com.ibm.ws.webcontainer/resources/OSGI-INF/metatype/metatype.xml

jhanders34 commented:
#libby

teddyjtorres reviewed (dismissed):
MP JWT 2.1 and Jakarta Security 3.0 changes look good.

cbridgha reviewed (approved):
Looks good

jhanders34 commented:
#libby
